torchvision/models/feature_extraction.py

```diff
@@ -0,0 +1,574 @@
+        elif not mode and self.training:
+            self.graph = self.eval_graph
+        return super().train(mode=mode)
+
+
+def build_feature_graph_net(
+        model: nn.Module,
+        return_nodes: Optional[Union[List[str], Dict[str, str]]] = None,
+        train_return_nodes: Optional[Union[List[str], Dict[str, str]]] = None,
+        eval_return_nodes: Optional[Union[List[str], Dict[str, str]]] = None,
+        tracer_kwargs: Dict = {}) -> fx.GraphModule:
+    """
+    Creates a new graph module that returns intermediate nodes from a given
+    model as dictionary with user specified keys as strings, and the requested
+    outputs as values. This is achieved by re-writing the computation graph of
+    the model via FX to return the desired nodes as outputs. All unused nodes
+    are removed, together with their corresponding parameters.
+
+    A note on node specification: A node qualified name is specified as a `.`
+    seperated path walking the hierarchy from top level module down to leaf
+    operation or leaf module. For instance `blocks.5.3.bn1`. The keys of the
+    `return_nodes` argument should point to either a node's qualified name,
+    or some truncated version of it. For example, one could provide `blocks.5`
+    as a key, and the last node with that prefix will be selected.
+    :func:`print_graph_node_qualified_names` is a useful helper function
+    for getting a list of qualified node names of a model.
+
+    An attempt is made to keep all non-parametric properties of the original
+    model, but existing properties of the constructed `GraphModule` are not
+    overwritten.
+
+    Not all models will be FX traceable, although with some massaging they can
+    be made to cooperate. Here's a (not exhaustive) list of tips:
+
+        - If you don't need to trace through a particular, problematic
+          sub-module, turn it into a "leaf module" by passing a list of
+          `leaf_modules` as one of the `tracer_kwargs` (see example below). It
+          will not be traced through, but rather, the resulting graph will
+          hold a reference to that module's forward method.
+        - Likewise, you may turn functions into leaf functions by passing a
+          list of `autowrap_functions` as one of the `tracer_kwargs` (see
+          example below).
+        - Some inbuilt Python functions can be problematic. For instance,
+          `int` will raise an error during tracig. You may wrap them in your
+          own function and then pass that in `autowrap_functions` as one of
+          the `tracer_kwargs`.
+        - Use `torch.matmul(tensor_a, tensor_b)` instead of `tensor_a @
+          tensor_b`.
+
+    For further information on FX see the
+    `torch.fx documentation <https://pytorch.org/docs/stable/fx.html>`_.
+
+    Args:
+        model (nn.Module): model on which we will extract the features
+        return_nodes (Optional[Union[List[str], Dict[str, str]]]): either a `List`
```

Comment:
We should avoid writing type-annotations in the docstrings: they're usually much less readable than a simple plain English description, and annotations are targeted for machines, not humans. In this case this could just be something like

```suggestion
        return_nodes (list or dict, optional) either a `List`
```

and the details of the expected types can be described below. This is how we usually try to document stuff: https://pytorch.org/vision/stable/transforms.html#torchvision.transforms.RandomCrop

Reply:
I remember you raising this in the discussion at #2025. Not sure we reached an agreement to replace `List[str]` with `list` in the discussion. Personally I'm happy with the precise definition but I acknowledge there is a certain degree of subjectivity on what's readable.

Reply:
@datumbox I believe these are 2 unrelated discussions.

Our discussion in #2025 was about whether having type annotations or not. I'm still firmly against it, but consensus wasn't reached.

This here however is about how we specify the types of the parameters in docstrings. We should never ever use the type annotation syntax for that.

Reply:
@datumbox it looks like you reacted with a "thumbs down" emoji to my comment above; would you mind elaborating? What part do you disagree with?

Reply:
@NicolasHug I don't agree with your "never ever" statement but at the same time I don't think it's worth over analyzing it. Sometimes people agree to disagree and move on. :)

Reply:
Just making sure to acknowledge this. I had a look at the the other examples in the repo and found a bit of a mix of both approaches. So I'm happy to leave it the way I have it now, unless a clear decision is made. Resolving this for now. And thanks both for the feedback!

Reply:
would you mind pointing to the current occurrences of annotation syntax in the docstrings? If there are any we should update them

Reply:
@NicolasHug this was just with a random click through of ctrl+F "Args:" in my IDE. Found an example for you: https://github.com/pytorch/vision/blob/e95db7aa09ac012dfbdb5db8a24ba8e5804d55b5/torchvision/utils.py#L162

Reply:
@NicolasHug sorted